### ISSUE TITLE
Some server's not showing in-game

### ISSUE THREAD

**Issue by jonathanprl** [rlabrecque/Steamworks.NET#295]
NOTE: I'm not a game developer, but I have around 500 game servers in the Rust server list. I've being doing some research and believe a particular aspect of Steamworks is causing me some issues. Any insight would be great, but I understand if this is not the place to post this!

I have an issue where some dedicated servers are not displaying in-game. After weeks of testing, I believe the issue is down to this particular feature of `RequestInternetServerList` in `ISteamMatchmakingServers`:

> The initial result is a flat list of IP addresses to query, returned approximately in order of how close the game server is to the requesting client. 

When Rust calls this function, and returns a list of servers to the player, how do you determine exactly "how close" the game server is to the player? Do you simply ping the server? Or do you lookup GeoIP information?

We have some IP ranges where only about 30-50% of game servers will show up in-game. However the server is still joinable directly. For a server hosted in Washington DC, we have players in New York, Florida, Texas etc that can't see it. However, I can change the IP without physically moving the server and it will show up. All these players trying to see it in the list also have a very good ping to the server.

With a server we have in Singapore, all players in Singapore, Vietname, Thailand etc and see it without an issue. Players in India can not see it, despite having a very good ping.

**Comment by KelsamGames** [comment]
Steamworks.NET is a C#-based library "go-between" for C# applications and the Steamworks SDK. What you're looking for is support for the Steamworks SDK itself.

Steamworks SDK Support forums are here: https://steamcommunity.com/groups/steamworks/steamworks

Documentation for Steamworks SDK can be found here: https://partner.steamgames.com/doc/api/ISteamMatchmakingServers

**Comment by jonathanprl** [comment]
Thank you very much, sorry for posting here!

**Comment by Alloc86** [comment]
I know this might be the wrong place, but as you typically don't get answers on Steam's official discussion groups and people on here are experienced with Steam I suppose some input might not hurt here ;)

First of all, have you received any information on this issue @jonathanprl ?

What we experienced so far: We got the same issues with servers not always showing up. One thing is that Steam's master server list seems to prioritize servers that are closer, and servers especially overseas are often taking a *lot* longer to show up, if at all.
But also it seems some don't show up at all, even closer ones. So far I think we had little reports about those servers being in single-instance environments, i.e. IPs that only server a single instance of our server (and no other Steam based servers). But we definitely got quite a bunch of such reports especially from GSPs that run a lot of server instances on a single IP. I asked Valve a long time ago about whether they are limiting entries per IP or something but of course never got a reply :(

Just my 2 cents, and sorry for the long wait :)
